src/reporter/plugin-host.ts

```diff
@@ -192,4 +192,9 @@ export default class ReporterPluginHost {
     public async reportTaskDone (/* endTime, passed, warnings */): Promise<never> {
         throw new Error('Not implemented');
     }
+
+    // NOTE: It's an optional method
+    // public async reportWarnings (/* warnings */): Promise<never> {
```

Comment:
You can define an empty function to make this plugin method optional, so you won't have to check its existence in the reporter. `reportTestStart` was implemented in a bad way.